features/telemetry/adapters/nxos/src/main/java/org/opennms/netmgt/telemetry/adapters/nxos/NxosGpbParserUtil.java

```diff
@@ -0,0 +1,192 @@
+ * bothering about structure they may follow.
+ *
+ * @author cgorantla
+ */
+public class NxosGpbParserUtil {
+
+    /**
+     * Given a Telemetry Message and field name, get it's value as double.
+     * 
+     * @param   msg  {@link org.opennms.netmgt.telemetry.adapters.nxos.proto.TelemetryBis.Telemetry} message
+     * @param   name  field name
+     * @return  value field value
+     */
+    public static Double getValueAsDouble(TelemetryBis.Telemetry msg, String name) {
+        for (TelemetryBis.TelemetryField field : msg.getDataGpbkvList()) {
+            TelemetryBis.TelemetryField subField = findFieldWithName(field, name);
+            if (subField != null) {
+                return getValue(subField);
+            }
+        }
+        return Double.NaN;
+    }
+
+    /**
+     * Given a Telemetry Message and field name, get it's value as string.
+     * 
+     * @param   msg  {@link org.opennms.netmgt.telemetry.adapters.nxos.proto.TelemetryBis.Telemetry} message
+     * @param   name  field name
+     * @return  value field value
+     */
+    public static String getValueAsString(TelemetryBis.Telemetry msg, String name) {
+        for (TelemetryBis.TelemetryField field : msg.getDataGpbkvList()) {
+            TelemetryBis.TelemetryField subField = findFieldWithName(field, name);
+            if (subField != null) {
+                return subField.getStringValue();
+            }
+        }
+        return "";
+    }
+
+    /**
+     * Given a Telemetry Message and row name, get the list of fields
+     * 
+     * @param   Telemetry message {@link org.opennms.netmgt.telemetry.adapters.nxos.proto.TelemetryBis.Telemetry}
+     * @param   name of field
+     * @return  list of Telemetry fields
+     */
+    public static List<TelemetryBis.TelemetryField> getRowsFromTable(Telemetry msg, String name) {
+        Objects.requireNonNull(msg);
+        for (TelemetryBis.TelemetryField field : msg.getDataGpbkvList()) {
+            // ROWS are arrays for a set of metrics, Assumption is that they
+            // start with ROW_
+            TelemetryBis.TelemetryField subField = findFieldWithName(field, "ROW_" + name);
+            if (subField != null) {
+                return subField.getFieldsList();
+            }
+        }
+        return null;
+
+    }
+
+    /**
+     * Given a Telemetry field row and name of field, get field value as String
+     * 
+     * @param   row {@link org.opennms.netmgt.telemetry.adapters.nxos.proto.TelemetryBis.TelemetryField}
+     * @param   name  field name
+     * @return  value field value
+     */
+    public static String getValueFromRowAsString(TelemetryBis.TelemetryField row, String name) {
+        for (TelemetryBis.TelemetryField nestedField : row.getFieldsList()) {
+            TelemetryBis.TelemetryField subField = findFieldWithName(nestedField, name);
+            if (subField != null) {
+                return subField.getStringValue();
+            }
+        }
+        return "";
```

Comment:
If we change the default value of `getValueAsString` we should change this one too.